### PR TITLE
Updating genetics portal parser

### PR DESCRIPTION
**Pull request by DSuveges**
This is the updated genetics portal evidence parser. There are quite many changes:

* The data preparation step is now removed, as that step is combined with the evidence generation
* README updated

**How to run:**

1. Create a cluster:
```bash
gcloud beta dataproc clusters create \
    ds-cluster-l2g-data \
    --image-version=1.4 \
    --properties=spark:spark.debug.maxToStringFields=100,spark:spark.executor.cores=31,spark:spark.executor.instances=1 \
    --master-machine-type=n1-standard-32 \
    --master-boot-disk-size=1TB \
    --zone=europe-west1-d \
    --single-node \
    --max-idle=20m \
    --region europe-west1 \
    --project=open-targets-genetics 
```
Once the dataproc cluster is done, submit job:
```bash
gcloud dataproc jobs submit pyspark \
     --cluster=ds-cluster-l2g-data \
    --project=open-targets-genetics \
     --region=europe-west1 \
    /Users/dsuveges/repositories/evidence_datasource_parsers/modules/GeneticsPortal_prepare_data.py -- \
    --locus2gene gs://genetics-portal-data/l2g/200127 \
    --toploci gs://genetics-portal-data/v2d/200207/toploci.parquet \
    --study gs://genetics-portal-data/v2d/200207/studies.parquet \
    --threshold 0.05 \
    --variantIndex gs://genetics-portal-data/variant-annotation/190129/variant-annotation.parquet  \
    --ecoCodes gs://genetics-portal-data/lut/vep_consequences.tsv \
    --output gs://genetics-portal-analysis/l2g-platform-export/data/genetics_portal_evidence.json.gz
```

### CODE REVIEW

**Comment by DSuveges** [comment]
This is an example of the generated evidence:

```json
{
  "datasourceId": "ot_genetics_portal",
  "datatypeId": "genetic_association",
  "targetFromSourceId": "ENSG00000163083",
  "diseaseFromSourceMappedId": "EFO_0004875",
  "literature": [
    "30038396"
  ],
  "publicationFirstAuthor": "Lee JJ",
  "publicationYear": 2018,
  "diseaseFromSource": "Self-reported math ability (MTAG) [MTAG]",
  "studyId": "GCST006569",
  "studySampleSize": 670471,
  "pValueMantissa": 2,
  "pValueExponent": -9,
  "resourceScore": 0.17988596856594086,
  "variantRsId": "rs13019313",
  "variantId": "2_120316287_A_G",
  "variantFunctionalConsequenceId": "SO_0001628"
}
```

It's validated successfully against current json schema.

File: README.md [hunk 1]
```diff
@@ -164,34 +164,23 @@ The version in _master_ is an older version (October 2018) that **DOES NOT** cal
...
-    --variantIndex=gs://genetics-portal-data/variant-annotation/190129/variant-annotation.parquet \
-    --ecoCodes=gs://genetics-portal-data/lut/vep_consequences.tsv \
-    --output=gs://genetics-portal-analysis/l2g-platform-export/data/l2g_joined.2020.03.02_exploded.parquet
+python GeneticsPortal.py \
+    --locus2gene gs://genetics-portal-data/l2g/200127 \
+    --toploci gs://genetics-portal-data/v2d/200207/toploci.parquet \
+    --study gs://genetics-portal-data/v2d/200207/studies.parquet \
+    --variantIndex gs://genetics-portal-data/variant-annotation/190129/variant-annotation.parquet \
+    --ecoCodes gs://genetics-portal-data/lut/vep_consequences.tsv \
+    --output gentics-portal-evidence.json.gz \
```
Comment (ireneisdoomed) [inline README.md:176]: Correct to `outputFile`.

File: modules/GeneticsPortal.py [hunk 2]
```diff
@@ -1,528 +1,261 @@
...
+        # multiple EFOs! We need to decide a strategy to deal with these.
+        
+        # # For example, only keep studies with 1 efo:
+        # .filter(size(col('trait_efos')) == 1)
+        # .withColumn('efo', col('trait_efos').getItem(0))
+        # .drop('trait_efos')
+
+        # Or, drop rows with no EFO and then explode array to multiple rows
+        .withColumn('trait_efos', when(col('trait_efos').isNotNull(),
+                                       expr('filter(trait_efos, t -> length(t) > 0)')))
```
Comment (ireneisdoomed) [inline modules/GeneticsPortal.py:146]: I would omit this piece of code. This new column function filters those arrays in trait_efos with a size of >= 0 whenever the value is not null.
What is the difference with line 147? There are 753 studies with an empty array and we're not interested in such cases.

**Review by ireneisdoomed — CHANGES_REQUESTED** [review]
I have ran it on a Dataproc job and everything looks fine. For a second iter I would suggest:
1.  to revisit the structure of the parser as a whole, making it more modular to not have everything on the main() function.
2. to review how the ECO dictionaries are imported. Both of the dictionaries share the key, I think it may make sense to just create one and have one function building the most severe consequence and the link altogether in a clearer way.

File: modules/GeneticsPortal.py (hunk 2 above)
Comment (DSuveges) [reply modules/GeneticsPortal.py:146]: It took a while for me to figure out what's going on here (first of all, you are right, row 147 is not needed). So we need to apply an extra filter, as there are 12 studies with empty string as efo. Where the explosion would still generated rows with no efo. You can try it out. Anyway, I'll simplify the code as much as I can and do an other commit.

```
+--------------------+--------------------+-----------+------------+
|            study_id|      trait_reported|sample_size|         efo|
+--------------------+--------------------+-----------+------------+
|          GCST008760|Perceived sweetne...|        686|            |
|          GCST008311|Lymphangioleiomyo...|       1278|            |
|          GCST008870|Keratinocyte canc...|     358840|            |
|     NEALE2_3761_raw|Age hay fever, rh...|      72232|            |
|          GCST008765|Perceived intensi...|       1757|            |
|          GCST008766|Perceived intensi...|       1757|            |
|          GCST009418|Lactose tolerance...|         60|            |
|          GCST008767|Perceived intensi...|        686|            |
|          GCST008763|Perceived intensi...|       1757|            |
|          GCST008768|Perceived intensi...|       2443|            |
|          GCST008769|Perceived intensi...|       1757|            |
|          GCST008764|Perceived intensi...|       1757|            |
```

File: README.md [hunk 3]
```diff
@@ -164,34 +164,23 @@ The version in _master_ is an older version (October 2018) that **DOES NOT** cal
...
-    --project=${projectName} \
-    --region=${region} \
-    ${repo_directory}/modules/GeneticsPortal_prepare_data.py -- \
-    --locus2gene=gs://genetics-portal-data/l2g/200127 \
-    --toploci=gs://genetics-portal-data/v2d/200207/toploci.parquet \
-    --study=gs://genetics-portal-data/v2d/200207/studies.parquet \
-    --variantIndex=gs://genetics-portal-data/variant-annotation/190129/variant-annotation.parquet \
-    --ecoCodes=gs://genetics-portal-data/lut/vep_consequences.tsv \
-    --output=gs://genetics-portal-analysis/l2g-platform-export/data/l2g_joined.2020.03.02_exploded.parquet
+python GeneticsPortal.py \
```
Comment (ireneisdoomed) [inline README.md:170]: Correct to `modules/GeneticsPortal.py`

Comment (DSuveges) [reply README.md]: Done.

File: modules/GeneticsPortal.py (hunk 2 above)
Comment (ireneisdoomed) [reply modules/GeneticsPortal.py:146]: I don't think I'm understanding this correctly: by keeping this piece of code and not the previous one:
```
.filter(col('trait_efos').isNotNull() & (size(col('trait_efos')) > 0))
```
you are keeping those rows where the `trait_efos` array is empty. So the remaining code contradicts with your comment above. Can you explain where I'm getting things wrong?

Comment (DSuveges) [reply modules/GeneticsPortal.py:146]: So in my understanding that expression filters for rows where the `trait_efos` value is not null and the length of the value is greater than zero, filtering out values where the only element is an empty string eg:
```
[ "" ]
```

**Review by ireneisdoomed — APPROVED** [review]

**Pull request merged**